src/Castle.Core/DynamicProxy/ProxyUtil.cs

```diff
@@ -15,15 +15,56 @@
 namespace Castle.DynamicProxy
 {
 	using System;
+	using System.Collections.Generic;
+	using System.Linq;
 	using System.Reflection;
+	using System.Runtime.CompilerServices;
 
-	using Castle.DynamicProxy.Internal;
 #if FEATURE_REMOTING
 	using System.Runtime.Remoting;
 #endif
 
+	using Castle.Core.Internal;
+	using Castle.DynamicProxy.Generators.Emitters;
+
 	public class ProxyUtil
 	{
+		private static readonly IDictionary<Assembly, bool> internalsVisibleToDynamicProxy = new Dictionary<Assembly, bool>();
+		private static readonly Lock internalsVisibleToDynamicProxyLock = Lock.Create();
+
+		/// <summary>
+		///   Determines whether this assembly has internals visible to dynamic proxy.
+		/// </summary>
+		/// <param name = "asm">The assembly to inspect.</param>
+		internal static bool AreInternalsVisibleToDynamicProxy(Assembly asm)
```

Comment:
Could you move this internal function further down, with the methods for accessibility.